exercises/risk-risiko.cpp

```diff
@@ -27,3 +27,215 @@
   M 3 vs 3 => blue win
   O 2 vs 1 => red win
 */
+
+#include <iostream>
+#include <cstdlib>
+#include <ctime>
+using namespace std;
+class Dice
+{
+private:
+  short n_sides;
+  short thrown_value;
+
+public:
+  Dice() : n_sides(6){};
+  short diceThrow()
+  {
+    thrown_value = rand() % n_sides + 1;
+    return thrown_value;
+  };
+  short const getValue() { return thrown_value; };
+  short const getSides() { return n_sides; };
+};
+
+class PlayerDices
+{
+private:
+  Dice *Dice1;
+  Dice *Dice2;
+  Dice *Dice3;
+  short N, M, O;
+
+public:
+  PlayerDices()
+  {
+    Dice1 = new Dice;
+    Dice2 = new Dice;
+    Dice3 = new Dice;
+  };
+  void tripleDiceThrow()
+  {
+    N = Dice1->diceThrow();
+    M = Dice2->diceThrow();
+    O = Dice3->diceThrow();
+
+    // N M O control
+    if (O >= N)
+      swap(O, N);
+    else if (M >= N)
+      swap(M, N);
+    if (O >= M)
+      swap(O, M);
+    else if (M >= N)
+      swap(M, N);
+
+    cout << N << " (N)" << endl;
+    cout << M << " (M)" << endl;
+    cout << O << " (O)" << endl;
+  }
+  bool compareN(short enemyN) // false/0 = defender, true/1 = attacker
+  {
+    if (this->N <= enemyN)
+      return false;
+    else
+      return true;
+  }
+  bool compareM(short enemyM) // false/0 = defender, true/1 = attacker
+  {
+    if (this->M <= enemyM)
+      return false;
+    else
+      return true;
+  }
+  bool compareO(short enemyO) // false/0 = defender, true/1 = attacker
+  {
+    if (this->O <= enemyO)
+      return false;
+    else
+      return true;
+  }
+  friend swap();
+  short const getN() { return N; };
+  short const getM() { return M; };
+  short const getO() { return O; };
+};
+
+void swap(short &a, short &b)
+{
+  short temp = a;
+  a = b;
+  b = temp;
+}
+
+bool coin_flip()
+{
+  // false/0 = defender, true/1 = attacker
+  string choice;
+  do
+  {
+    cout << "Heads or tails?" << endl;
+    cin >> choice;
+    for (int i = 0; i < choice.length(); i++) // formatting the input, just in case someone types something like "TaIlS"
+      choice[i] = tolower(choice[i]);
+
+    bool coin = rand() % 2; // heads = 0, tails = 1
+    cout << endl;
+    if (coin == 0)
+      cout << "It's heads!" << endl;
+    else
+      cout << "It's tails!" << endl;
+
+    if (choice == "heads" && coin == 0)
+    {
+      cout << "You're the attacker! (Red)" << endl
+           << endl;
+      return true;
+    }
+    else if (choice == "tails" && coin == 1)
+    {
+      cout << "You're the attacker! (Red)" << endl
+           << endl;
+      return true;
+    }
+    else if (choice != "heads" && choice != "tails")
+    {
+      cout << "But you wrote neither heads nor tails... try again." << endl;
+      continue;
+    }
+    else
+    {
+      cout << "You're the defender! (Blue)" << endl
+           << endl;
+    }
+    return false;
```

Comment:
Gli else if li puoi convertire in if mettendo anche dei ritorni a capo tra loro

Essendoci return o continue quell' else non serve